gateway-service/src/main/java/org/zowe/apiml/gateway/config/AdditionalRegistrationConfig.java

```diff
@@ -0,0 +1,132 @@
+        return additionalRegistrations;
+    }
+
+    static List<AdditionalRegistration> extractAdditionalRegistrations(Map<String, String> allProperties) {
+
+        if (CollectionUtils.isEmpty(allProperties)) {
+            return Collections.emptyList();
+        }
+
+        Map<String, String> additionalProperties = allProperties.entrySet().stream().filter(entry -> entry.getKey().startsWith(COMMON_PREFIX))
+            .filter(entry -> StringUtils.isNotBlank(entry.getValue()))
+            .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().replace(COMMON_PREFIX, ""), entry.getValue()))
+            .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue, (previous, current) -> current, TreeMap::new));
+
+        for (Map.Entry<String, String> prp : additionalProperties.entrySet()) {
+            log.debug("Additional property: {}={}", prp.getKey(), prp.getValue());
+        }
+
+        Integer listSize = additionalProperties.keySet().stream().map(key -> parsePropertyName(key).getKey())
+            .max(Integer::compareTo).map(maxIndex -> maxIndex + 1).orElse(0);
+
+        List<AdditionalRegistration> additionalRegistrations = IntStream.range(0, listSize)
+            .mapToObj(index -> AdditionalRegistration.builder().routes(new ArrayList<>()).build())
+            .collect(Collectors.toList());
+
+        mapProperties(additionalRegistrations, additionalProperties);
+        for (AdditionalRegistration registration : additionalRegistrations) {
+            List<AdditionalRegistration.Route> definedRoutes = registration.getRoutes().stream().filter(route -> !AdditionalRegistrationConfig.isRouteDefined(route)).collect(Collectors.toList());
+            registration.setRoutes(definedRoutes);
+        }
+
+        return additionalRegistrations.stream()
+            .filter(registration -> StringUtils.isNotBlank(registration.getDiscoveryServiceUrls()))
+            .collect(Collectors.toList());
+    }
+
+    private static void mapProperties(List<AdditionalRegistration> additionalRegistrations, Map<String, String> properties) {
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String propertyName = entry.getKey();
+            Pair<Integer, String> property = parsePropertyName(propertyName);
+            final String propertyValue = entry.getValue();
+            switch (StringUtils.upperCase(property.getValue())) {
+                case DISCOVERY_SERVICE_URLS_KEY:
+                    additionalRegistrations.get(property.getKey()).setDiscoveryServiceUrls(propertyValue);
+                    break;
+                case GATEWAY_URL_KEY:
+                    setRouteProperty(additionalRegistrations.get(property.getKey()), parseRouteIndex(propertyName), (AdditionalRegistration.Route route) -> route.setGatewayUrl(propertyValue));
```

Comment:
I think that we should provide some default value for routes, for now there is other option then "/" and probably will not be so it would be easier for users to setup if there is default value

Reply:
updated